### ISSUE TITLE
Benchmarks don't seem to exercise collision handling

### ISSUE THREAD

**Issue by sjakobi** [haskell-unordered-containers/unordered-containers#439]
We currently have benchmarks with keys of type `Int`, `ByteString` and `String`. All of these use the instances provided in `Hashable` which I'd expect not to produce a lot of hash collisions.

For tuning the code that handles collisions, e.g. https://github.com/haskell-unordered-containers/unordered-containers/pull/435, it would be useful to have a key type that is more collision-prone.

**Comment by treeowl** [comment]
Historically, the view has been that collisions had better be rare enough not to matter much. There's plenty of O(n^2) gunk in collision buckets.

**Comment by sjakobi** [comment]
Yeah, this isn't a high priority task. It's mostly the result of me wanting to try the `searchSwap` trick in #435. Now I simply don't have any benchmarks to tell whether the new code is actually better or not. ;)
